### DISCUSSION TITLE
Should we just not revert when the array is empty in `withdrawMultiple`?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#50] [Ideas]
There would be a few benefits:

1. Reduced gas cost (due to there being one if statement fewer)
2. Easier to maintain the tests (due to there being one testing branch fewer)

I can't think of any major downside.

**Comment by razgraf** [comment]
Looks fine, empty arrays would probably be due to UI bugs, which shouldn't happen too often. The worse they'll do is consume some gas for the non-state-altering tx. Maybe it's worth looking into what others do when expecting this type of arrays as params?

**Reply by PaulRBerg** [reply]
Yeah I'll try to find an equivalent scenario in OpenZeppelin's code base.

**Reply by PaulRBerg** [reply]
@andreivladbrg it would be good to spend a little bit of time to see what OpenZeppelin does in this case.

**Comment by PaulRBerg** [comment]
@andreivladbrg could you please research what OpenZeppelin does with zero-length arrays so we can make a decision about this?

**Reply by andreivladbrg** [reply]
I didn't find a pattern in their contracts where they revert when an array has a zero length.

Anyway, the discussion is outdated because we no longer check for this in the "multiple" functions.

**Reply by PaulRBerg** [reply]
Yup, you're right, this discussion no longer applies.

Should we add tests for when the array length is zero?

**Reply by andreivladbrg** [reply]
We could

**Reply by PaulRBerg** [reply]
We will make a final decision on this topic once we make up our minds regarding the proxy (https://github.com/sablierhq/v2-periphery/discussions/11).

**Reply by PaulRBerg** [reply]
@andreivladbrg I ended up writing a test in PR https://github.com/sablierhq/v2-core/pull/341 for the case when the length of the array is zero in `withdrawMultiple`.

Once that PR is merged in, we can lock this discussion for good.

**Reply by PaulRBerg** [reply]
Locking, as #341 has been merged.
